### PR TITLE
Add latest release tag to be the prefix of latest version of Chart

### PR DESCRIPTION
**Pull request by zzxwill**
Currently the latest chart is vela-core:latest, which could
not be supported by Artifacthub, add latest tag and make the
lastest chart version as `vx.y.z-latest`

### CODE REVIEW

File: .github/workflows/registry.yml [hunk 1]
```diff
@@ -92,14 +84,24 @@ jobs:
...
-          sed -i "s/latest/$version/g" $LEGACY_HELM_CHART/values.yaml
-          number=${version#"v"}
-          sed -i "s/0.1.0/$number/g" $HELM_CHART/Chart.yaml
-          sed -i "s/0.1.0/$number/g" $LEGACY_HELM_CHART/Chart.yaml
+          latest_tag=${{ steps.latest_tag.outputs.tag }}
+          tag=${GITHUB_REF#refs/tags/}
+          chart_version=${tag#"v"}
+          if [[ ${GITHUB_REF} == "refs/heads/master" ]]; then
+            tag=latest
+            chart_version=${latest_tag#"v"}-latest
```
Comment (wonderflow) [inline .github/workflows/registry.yml:99]: great job!  Let's do it better:
1. the latest version, let's add one for the minor version. for example, the current version is `0.3.2` then, we should  use `0.4.0-latest-master`
2. make sure the user can only install it by using the flag `--devel`
3. align the installation doc

Comment (zzxwill) [reply .github/workflows/registry.yml]: Done.

File: .github/workflows/registry.yml [hunk 2]
```diff
@@ -93,14 +85,29 @@ jobs:
...
-          version=${{ steps.get_version.outputs.VERSION }}
-          sed -i "s/latest/$version/g" $HELM_CHART/values.yaml
-          sed -i "s/latest/$version/g" $LEGACY_HELM_CHART/values.yaml
-          number=${version#"v"}
-          sed -i "s/0.1.0/$number/g" $HELM_CHART/Chart.yaml
-          sed -i "s/0.1.0/$number/g" $LEGACY_HELM_CHART/Chart.yaml
+          latest_tag=${{ steps.latest_tag.outputs.tag }}
+          sub="."
+          major="$(cut -d"$sub" -f1 <<<"$latest_tag")"
+          minor="$(cut -d"$sub" -f2 <<<"$latest_tag")"
```
Comment (wonderflow) [inline .github/workflows/registry.yml:98]: what will happen if the cut failed, for example the latest tag become v1.0 for some reason.

File: docs/en/install.md [hunk 3]
```diff
@@ -104,9 +101,21 @@ These steps will install KubeVela controller and its dependency.
     helm install -n vela-system kubevela kubevela/vela-core --set installCertManager=true
     ```
    
-    If you want to try the latest master branch, try the following command:
+    If you want to try the latest master branch, add flag `--devel` in command `helm search` to choose the latest
```
Comment (wonderflow) [inline docs/en/install.md:104]: how does the helm distinguish a chart can be normally install or must use `--devel`?  What is the parttern? I guess a version with a suffix will cause it into `--devel`? Am I right?

File: docs/en/install.md [hunk 4]
```diff
@@ -104,9 +101,21 @@ These steps will install KubeVela controller and its dependency.
...
+    pre-release versions, like `0.3.2-latest-master`.
+   
+    ```shell script
+    $ helm search repo kubevela/vela-core -l --devel
+    NAME                     	CHART VERSION        	APP VERSION          	DESCRIPTION
+    kubevela/vela-core       	0.3.2-latest-master  	0.3.2-latest-master  	A Helm chart for Kube Vela core
+    poc/vela-core       	    0.3.2        	        0.3.2      	            A Helm chart for Kube Vela core
+    poc/vela-core       	    0.3.1        	        0.3.1               	A Helm chart for Kube Vela core
```
Comment (wonderflow) [inline docs/en/install.md:112]: what does `poc/vela-core` mean?

File: docs/en/install.md [hunk 5]
```diff
@@ -104,9 +101,21 @@ These steps will install KubeVela controller and its dependency.
     helm install -n vela-system kubevela kubevela/vela-core --set installCertManager=true
     ```
    
-    If you want to try the latest master branch, try the following command:
+    If you want to try the latest master branch, add flag `--devel` in command `helm search` to choose the latest
+    pre-release versions, like `0.3.2-latest-master`.
```
Comment (wonderflow) [inline docs/en/install.md:105]: maybe we need to expain more to tell how the `0.3.2-latest-master` generated? And what does it mean?

File: docs/en/install.md [hunk 6]
```diff
@@ -104,9 +101,21 @@ These steps will install KubeVela controller and its dependency.
...
+    kubevela/vela-core       	0.3.2-latest-master  	0.3.2-latest-master  	A Helm chart for Kube Vela core
+    poc/vela-core       	    0.3.2        	        0.3.2      	            A Helm chart for Kube Vela core
+    poc/vela-core       	    0.3.1        	        0.3.1               	A Helm chart for Kube Vela core
+    ```
+   
+    And try the following command to install it.
+   
     ```shell script
-    helm install -n vela-system kubevela  https://kubevelacharts.oss-cn-hangzhou.aliyuncs.com/core/vela-core-latest.tgz --set useWebhook=false
+    helm install -n vela-system kubevela kubevela/vela-core --version 0.3.2-latest-master --set useWebhook=false
```
Comment (wonderflow) [inline docs/en/install.md:118]:
```suggestion
    helm install -n vela-system kubevela kubevela/vela-core --version <next_version>-latest-master --set useWebhook=false
```

File: .github/workflows/registry.yml (hunk 2 above)
Comment (zzxwill) [reply .github/workflows/registry.yml:98]: All tags should follow the rule of `major.minor.patch` or `major.minor` if we wanna to build a valid helm chart. If the `cut` failed, it's

File: docs/en/install.md (hunk 3 above)
Comment (zzxwill) [reply docs/en/install.md:104]: Your guessing is correct. `--devel` means `use development versions (alpha, beta, and release candidate releases), too. Equivalent to version '>0.0.0-0'. If --version is set, this is ignored`.

For my local PoC, all these tags will display with `--devel`  flag.
```
kubevela/vela-core       	0.4.0-rc1.11.g4d8be8c	0.4.0-rc1.11.g4d8be8c	A Helm chart for Kube Vela core
kubevela/vela-core       	0.4.0-latest-master  	0.4.0-latest-master  	A Helm chart for Kube Vela core
kubevela/vela-core       	0.0.3-rc.1           	0.0.3-rc.1           	A Helm chart for Kube Vela core
kubevela/vela-core       	0.0.3-rc             	0.0.3-rc             	A Helm chart for Kube Vela core
```

File: docs/en/install.md (hunk 4 above)
Comment (zzxwill) [reply docs/en/install.md:112]: My fault.

File: docs/en/install.md (hunk 5 above)
Comment (zzxwill) [reply docs/en/install.md:105]: Then we should use version like `v0.3.2-rc` which means the current release is `v0.3.1`, the future one is `v0.3.2`  and this one if the candidate release for `v0.3.2`. With this pattern, I think no doc is needed.

**Review by wonderflow — APPROVED** [review]
Cool! Feel free to merge when you have checked in your own repo.

**Pull request merged**